packages/shared/src/hooks/useReadingRank.ts

```diff
@@ -29,15 +30,27 @@ const defaultRank: MyRankData = {
   reads: 0,
 };
 
+const checkShouldShowRankModal = (rankLastSeen: Date) => {
+  if (!rankLastSeen) {
+    return true;
+  }
+
+  return new Date(rankLastSeen) < new Date();
+};
+
 export default function useReadingRank(): ReturnType {
+  const { alerts, updateAlerts } = useContext(AlertContext);
   const { user, tokenRefreshed } = useContext(AuthContext);
   const [levelUp, setLevelUp] = useState(false);
   const queryClient = useQueryClient();
 
   const [cachedRank, setCachedRank, loadedCache] = usePersistentState<
-    MyRankData & { userId: string; neverShowRankModal: boolean }
+    MyRankData & { userId: string; neverShowRankModal?: boolean }
   >('rank', null);
   const neverShowRankModal = cachedRank?.neverShowRankModal;
+  const shouldShowRankModal = user
```

Comment:
I think that in any way we should first prioritize the value in the cache `neverShowRankModal` and only then fallback to the remote settings